compiler/rustc_target/src/asm/xtensa.rs

```diff
@@ -0,0 +1,244 @@
+use super::{InlineAsmArch, InlineAsmType};
+use crate::spec::Target;
+use rustc_macros::HashStable_Generic;
+use std::fmt;
+
+def_reg_class! {
+    Xtensa XtensaInlineAsmRegClass {
+        reg,
+        freg,
+        breg,
+    }
+}
+
+impl XtensaInlineAsmRegClass {
+    pub fn valid_modifiers(self, _arch: super::InlineAsmArch) -> &'static [char] {
+        &[]
+    }
+
+    pub fn suggest_class(self, _arch: InlineAsmArch, _ty: InlineAsmType) -> Option<Self> {
+        None
+    }
+
+    pub fn suggest_modifier(
+        self,
+        _arch: InlineAsmArch,
+        _ty: InlineAsmType,
+    ) -> Option<(char, &'static str)> {
+        None
+    }
+
+    pub fn default_modifier(self, _arch: InlineAsmArch) -> Option<(char, &'static str)> {
+        None
+    }
+
+    pub fn supported_types(
+        self,
+        _arch: InlineAsmArch,
+    ) -> &'static [(InlineAsmType, Option<&'static str>)] {
+        match self {
+            Self::reg | Self::breg => types! { _: I8, I16, I32; },
```

Comment:
Do you think this might be needed to limit based on feature "bool".

```Rust
            Self::reg => types! { _: I8, I16, I32; },
            Self::breg => types! { "bool": I8, I16, I32; },
```

Reply:
Well technically breg is the size of `i1`, but Rust doesn't seem to support that.

Reply:
I'm thinking more along the lines of breg only being available if the "bool" feature is available, matching the register that are limited by the feature.

Reply:
Adding support for is was very simple actually - added in latest commit.